app/views/edit/build-config.html

```diff
@@ -20,57 +20,90 @@
+                                    id="sourceContextDir"
+                                    name="sourceContextDir"
+                                    type="text"
+                                    ng-model="updatedBuildConfig.spec.source.contextDir"
+                                    placeholder="/"
+                                    autocorrect="off"
+                                    autocapitalize="off"
+                                    spellcheck="false">
+                                </div>
                               </div>
-                            </div>
 
-                            <div class="form-group">
-                              <label for="sourceContextDir">Source Context Dir</label>
-                              <div>
-                                <input class="form-control"
-                                  id="sourceContextDir"
-                                  name="sourceContextDir"
-                                  type="text"
-                                  ng-model="updatedBuildConfig.spec.source.contextDir"
-                                  placeholder="/"
-                                  autocorrect="off"
-                                  autocapitalize="off"
-                                  spellcheck="false">
+                              <div class="form-group">
+                                <osc-secrets model="secrets.picked.gitSecret" 
+                                            namespace="projectName"
+                                            type="Git Secret"
+                                            all-secrets="secrets.sourceManipulation"
+                                            alerts-callback="getAlerts()"
+                                            load-secrets-callback="getSecrets()">
+                                </osc-secrets>
+                              </div>
+
+                              <div class="form-group" ng-if="strategyType === 'Docker' || strategyType === 'Source' || strategyType === 'Custom'">
+                                <osc-secrets model="secrets.picked.sourceSecrets" 
+                                            namespace="projectName"
+                                            type="Source Secret"
+                                            all-secrets="secrets.sourceManipulation"
+                                            additional-input-field-name="{{strategyType === 'Custom' ? 'mountPath' : 'destinationDir'}}"
+                                            additional-input-field-tooltip="{{strategyType === 'Custom' ? 'Mount Path is the path at which to mount the secret in the build pod.' : 'Destination Dir is the directory where the files from the secret should be available for the build time.'}}"
+                                            alerts-callback="getAlerts()"
+                                            load-secrets-callback="getSecrets()">
+                                </osc-secrets>
```

Comment:
I don't understand why we have both "Git Secret" and "Source Secret". Shouldn't we only have one?